components/board/fake/board.go

```diff
@@ -233,39 +238,51 @@ func (b *Board) SetPowerMode(ctx context.Context, mode pb.PowerMode, duration *t
 func (b *Board) StreamTicks(ctx context.Context, interrupts []board.DigitalInterrupt, ch chan board.Tick,
 	extra map[string]interface{},
 ) error {
-	for _, i := range interrupts {
-		name := i.Name()
-		d, ok := b.Digitals[name]
+	for _, di := range interrupts {
+		_, ok := b.Digitals[di.Name()]
 		if !ok {
-			return fmt.Errorf("could not find digital interrupt: %s", name)
-		}
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		default:
-			// Keep going
-		}
-		// Get a random bool for the high tick value.
-		// linter complans about security but we don't care if someone
-		// can predict if the fake interrupts will be high or low.
-		//nolint:gosec
-		randBool := rand.Int()%2 == 0
-		select {
-		case ch <- board.Tick{Name: d.conf.Name, High: randBool, TimestampNanosec: uint64(time.Now().Unix())}:
-		default:
-			// if nothing is listening to the channel just do nothing.
+			return fmt.Errorf("could not find digital interrupt: %s", di.Name())
 		}
 	}
+
+	for _, di := range interrupts {
+		// Don't need to check if interrupt exists, just did that above
+		b.workers.AddWorkers(func(workersContext context.Context) {
+			for {
+				// sleep to avoid a busy loop
+				if !utils.SelectContextOrWait(ctx, 700*time.Millisecond) {
```

Comment:
:eyes: You raised a good question earlier about what to do if we care about multiple contexts. Is there a way for `SelectContextOrWait` to care about multiple ones, too? If not, we might wait 700 ms before shutting down, if `workersContext` is canceled but `ctx` is not yet. Looking at https://github.com/viamrobotics/goutils/blob/main/runtime.go#L188, I don't think there's anything we can do about this. and we can't easily make the context argument variadic because only the last argument to a Go function can be variadic, and swapping the argument order across multiple repos is nearly impossible.

Reply:
...I think I'd prefer to use `workersContext` on this line, given that we can't use both. That way, we close the component quickly when we're supposed to, and if we're merely supposed to stop sending ticks, we might shut down the goroutine 700ms late but won't send any interrupts in the meantime (the behavior is still correct).

Reply:
(but I don't feel strongly. Feel free to merge as-is if you want.)

Reply:
you're right, worker context would be better here